### PR TITLE
[SPARK-19319][SparkR]:SparkR Kmeans summary returns error when the cluster size doesn't equal to k

### PR DESCRIPTION
**Pull request by wangmiao1981**
## What changes were proposed in this pull request

When Kmeans using initMode = "random" and some random seed, it is possible the actual cluster size doesn't equal to the configured `k`.

In this case, summary(model) returns error due to the number of cols of coefficient matrix doesn't equal to k.

Example:
>  col1 <- c(1, 2, 3, 4, 0, 1, 2, 3, 4, 0)
>   col2 <- c(1, 2, 3, 4, 0, 1, 2, 3, 4, 0)
>   col3 <- c(1, 2, 3, 4, 0, 1, 2, 3, 4, 0)
>   cols <- as.data.frame(cbind(col1, col2, col3))
>   df <- createDataFrame(cols)
> 
>   model2 <- spark.kmeans(data = df, ~ ., k = 5, maxIter = 10,  initMode = "random", seed = 22222, tol = 1E-5)
> 
> summary(model2)
Error in `colnames<-`(`*tmp*`, value = c("col1", "col2", "col3")) :
  length of 'dimnames' [2] not equal to array extent
In addition: Warning message:
In matrix(coefficients, ncol = k) :
  data length [9] is not a sub-multiple or multiple of the number of rows [2]

Fix: Get the actual cluster size in the summary and use it to build the coefficient matrix.
## How was this patch tested?

Add unit tests.

### CODE REVIEW

**Comment by felixcheung** [comment]
ah - does bisecting kmeans have the same behavior?

File: R/pkg/R/mllib_clustering.R [hunk 1]
```diff
@@ -225,10 +225,12 @@ setMethod("spark.kmeans", signature(data = "SparkDataFrame", formula = "formula"
...
 #' @param object a fitted k-means model.
 #' @return \code{summary} returns summary information of the fitted model, which is a list.
-#'         The list includes the model's \code{k} (number of cluster centers),
+#'         The list includes the model's \code{k} (the configured number of cluster centers),
 #'         \code{coefficients} (model cluster centers),
-#'         \code{size} (number of data points in each cluster), and \code{cluster}
-#'         (cluster centers of the transformed data).
+#'         \code{size} (number of data points in each cluster), \code{cluster}
+#'         (cluster centers of the transformed data), and \code{clusterSize}
+#'         (the actual number of cluster centers. When using initMode = "random",
```
Comment (felixcheung) [inline R/pkg/R/mllib_clustering.R]: let's add `is.loaded` here

Comment (wangmiao1981) [inline R/pkg/R/mllib_clustering.R]: OK. I will add it. For bisecting kmeans, I haven't found a case like this. This case only occurs when initMode is random and this behavior was due to one fix to kmeans implementation.

**Comment by wangmiao1981** [comment]
Jenkins, retest this please.

**Comment by felixcheung** [comment]
LGTM

**Comment by wangmiao1981** [comment]
ping @felixcheung

**Comment by felixcheung** [comment]
merged, thanks!
I think it'll good to have this in branch-2.1 - @wangmiao1981 would you by any chance would like to backport this fix?

**Comment by wangmiao1981** [comment]
I will backport it soon. Thanks!

**Pull request closed without merging**